### ISSUE TITLE
Support `torch.linalg.diagonal`

### ISSUE THREAD

**Issue by AnirudhDagar** [pytorch/pytorch#62813]
## 🚀 Feature
Support `torch.linalg.diagonal` alias to [torch.diagonal](https://pytorch.org/docs/stable/generated/torch.diagonal.html).

cc @mruberry @rgommers @pmeier @asmeurer @leofang @AnirudhDagar @asi1024 @emcastillo @kmaehashi @heitorschueroff

## Motivation & Pitch

Python Array API proposes the use of `linalg.diagonal` among other linear algebra functions. Currently, PyTorch supports the same functionality with `torch.diagonal`. Array-API Ops tracker (#58742) lists a few aliases and it would be nice to add an alias `torch.linalg.diagonal` to `torch.diagonal` making it compliant with Array API.

## Additional context

* [Python Array API linalg.diagonal](https://data-apis.org/array-api/latest/extensions/linear_algebra_functions.html#linalg-diagonal-x-axis1-0-axis2-1-offset-0)

**Comment by rgommers** [comment]
xref https://github.com/data-apis/array-api/issues/215 for the default value of `axis1`, `axis2`.

**Comment by mruberry** [comment]
This would be nice to do. It may be easier to create a new function in the linalg namespace if there are BC/FC issues with PyTorch's current diagonal.

**Comment by AnirudhDagar** [comment]
@rgommers this is now unblocked by the fix in data-apis/array-api#241 which removes `axis1` and `axis2` altogether from the signature. Right?

**Comment by rgommers** [comment]
The signature there is `linalg.diagonal(x, /, *, offset=0)`, and the spec says the diagonal is taken over the last two axes and batching is supported.

The current `torch.diagonal` has signature:
```
torch.diagonal(input, offset=0, dim1=0, dim2=1)
```
and has a note:
```
To take a batch diagonal, pass in dim1=-2, dim2=-1.
```
So it looks to me like that's still not great, and adding a `torch.linalg.diagonal` sounds like the way to go. This could either be identical to the signature of the array API standard, or it could add `dim1=-2, dim2=-1` if desired (that's still compatible with the standard). The current `torch.diagonal` default is less useful.
